std/archive/tar.ts

```diff
@@ -105,15 +105,16 @@ function pad(num: number, bytes: number, base?: number): string {
   return "000000000000".substr(numString.length + 12 - bytes) + numString;
 }
 
-const types: { [key: string]: string } = {
-  "": "file",
-  "0": "file",
-  "1": "link",
-  "2": "symlink",
-  "3": "character-device",
-  "4": "block-device",
-  "5": "directory",
-};
+enum Types {
+  "file" = 0,
+  "link" = 1,
+  "symlink" = 2,
+  "character-device" = 3,
+  "block-device" = 4,
+  "directory" = 5,
+  "fifo" = 6,
+  "contiguous-file" = 7,
+}
```

Comment:
If the value is only used as a string why not use a string here?

```ts
enum Types {
  "file" = "0",
  "link" = "1",
  // ...
}
```

Reply:
in method #getMetadata of Class Untar, convert a numeric string to string, such as "0" => "file"
```ts
// before
meta.type = types[meta.type as string] || meta.type;
// now
meta.type =
  FileTypes[meta.type as keyof typeof FileTypes] as unknown as string ??
    meta.type;
```  

if the value is string,  i get an error
```ts
enum FileTypes {
  "file" = "0",
  "link" = 1,
}
console.log(FileTypes.file);
// ok
console.log(FileTypes[1]);
// throw an error: Element implicitly has an 'any' type because index expression is not of type 'number'.ts(7015)
console.log(FileTypes["0"]);
```